### PR TITLE
Issue 2

### PR DESCRIPTION
**Pull request by h714714**

### CODE REVIEW

**Comment by h714714** [comment]
[Issue 2: Save Image Link](https://trello.com/c/Mya5un13/202-issue-2-save-image-link)

File: test/models/image_test.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+require 'test_helper'
+
+class ImageTest < ActiveSupport::TestCase
+  def test_empty_url
+    empty_url = Image.new()
+    assert_not empty_url.valid?
+  end
+  def test_invalid_url
+    invalid_url = Image.new(image_url:"123456")
```
Comment (eldonchen0501) [inline test/models/image_test.rb]: we usually add a space after the `:` when we assigned params to a function

File: test/models/image_test.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    assert_not invalid_url.valid?
+  end
+  def test_valid_url
+    valid_url = Image.new(image_url:"http://example.com")
```
Comment (eldonchen0501) [inline test/models/image_test.rb]: same here

File: app/controllers/images_controller.rb [hunk 3]
```diff
@@ -2,4 +2,17 @@ class ImagesController < ApplicationController
   def new
     @image = Image.new
   end
+
+  def create
+    puts params
```
Comment (eldonchen0501) [inline app/controllers/images_controller.rb]: let's get rid of this line

File: app/controllers/images_controller.rb [hunk 4]
```diff
@@ -2,4 +2,17 @@ class ImagesController < ApplicationController
...
+    @image = Image.new(image_url:params[:image][:image_url])
+
+    if @image.valid?
+      @image.save!
+      redirect_to "/images/#{@image.id}"
```
Comment (eldonchen0501) [inline app/controllers/images_controller.rb]: we can also use rails path helper here like what we did in test

File: test/controllers/ImagesController_test.rb [hunk 5]
```diff
@@ -5,4 +5,17 @@ def test_simple_test
     get root_path
     assert_response :ok
   end
+
+  def test_create_valid_url
+    assert_difference -> {Image.count} => 1 do
```
Comment (eldonchen0501) [inline test/controllers/ImagesController_test.rb]: normally, we'll use syntax like `assert_difference 'Image.count', 1 do` for assert_difference

File: test/controllers/ImagesController_test.rb [hunk 6]
```diff
@@ -18,4 +18,15 @@ def test_create_invalid_url
     end
   end
 
+  def test_show_views
+    image = Image.create!(image_url:'https://tineye.com/images/widgets/mona.jpg')
+    # get '/images/#{@image.id}'
```
Comment (eldonchen0501) [inline test/controllers/ImagesController_test.rb]: let's get rid of this line

File: test/models/image_test.rb (hunk 1 above)
Comment (eldonchen0501) [reply test/models/image_test.rb]: nvm, I just saw them fixed in later commit

File: app/controllers/images_controller.rb (hunk 3 above)
Comment (h714714) [reply app/controllers/images_controller.rb]: ok

File: test/controllers/ImagesController_test.rb (hunk 5 above)
Comment (h714714) [reply test/controllers/ImagesController_test.rb]: ok

**Pull request merged**